ClosedXML/Excel/CalcEngine/Functions/Lookup.cs

```diff
@@ -242,9 +250,15 @@ private static object Vlookup(List<Expression> p)
                 matching_row = range.FindRow(r =>
                 {
                     var row_index_in_range = r.RowNumber() - first_row + 1;
-                    if (row_index_in_range < range.RowsUsed().Count() && !r.Cell(1).IsEmpty() && new Expression(r.Cell(1).Value).CompareTo(lookup_value) <= 0 && !r.RowBelow().Cell(1).IsEmpty() && new Expression(r.RowBelow().Cell(1).Value).CompareTo(lookup_value) > 0)
+                    if (row_index_in_range < range.RowsUsed().Count()
```

Comment:
I have a strong suspicion, `range.RowsUsed().Count()` will be called on every iteration in `FindRow` which is not very efficient. Better calculate the value outside the lambda expression.

Reply:
Fixed in #1237